src/api/proxies/create-dispatch-proxy.js

```diff
@@ -0,0 +1,143 @@
+import * as Actions from 'api/actions';
+import { normalize } from 'normalizr';
+
+import {
+  getActionKeyFromArgs,
+  splitArgs,
+  displayError,
+  actionNameForCall,
+} from 'utils/api-helpers';
+
+export const createDispatchProxy = Provider => {
+  const client = new Provider();
+
+  return new Proxy(createDispatchProxy, {
```

Comment:
how would you code that? I need to intercept any getter called.
in this first proxy, this is the API namespace ("activity", "repository", "...") : `client.**activity**.something()`

Reply:
I am not sure. I just wonder it should be able to work. Maybe we should separate `Client` as utils and api config first.

Reply:
Keep in mind that the first Proxy is just for cosmetics. It enables us to have namespaces.
i.e. client.activity.getEventsReceived() instead of client.activityGetEventsReceived()